src/lib.rs

```diff
@@ -589,6 +733,114 @@ impl Contract {
         self.edit_proposal_internal(id, body.into(), proposal.snapshot.labels)
     }
 
+    #[payable]
+    pub fn edit_proposal_linked_rfp(&mut self, id: ProposalId, rfp_id: Option<RFPId>) -> Promise {
+        near_sdk::log!("edit_proposal_linked_rfp");
+        let proposal: Proposal = self
+            .proposals
+            .get(id.into())
+            .unwrap_or_else(|| panic!("Proposal id {} not found", id))
+            .into();
+        let mut body = proposal.snapshot.body.latest_version();
+        body.linked_rfp = rfp_id;
+
+        self.edit_proposal_internal(id, body.into(), proposal.snapshot.labels)
+    }
+
+    fn assert_can_link_unlink_rfp(&self, rfp_id: Option<RFPId>) {
+        if let Some(rfp_id) = rfp_id {
+            let rfp: RFP = self
+                .rfps
+                .get(rfp_id.into())
+                .unwrap_or_else(|| panic!("RFP id {} not found", rfp_id))
+                .into();
+            require!(
+                rfp.snapshot.body.latest_version().timeline.is_accepting_submissions() || self.is_allowed_to_write_rfps(env::predecessor_account_id()),
+                format!("The RFP {} is not in the Accepting Submissions state, so you can't link or unlink to this RFP", rfp_id)
+            );
+        }
+    }
+
+    fn get_rfp_labels(&self, rfp_id: RFPId) -> HashSet<String> {
+        let rfp: RFP = self
+            .rfps
+            .get(rfp_id.into())
+            .unwrap_or_else(|| panic!("RFP id {} not found", rfp_id))
+            .into();
+        rfp.snapshot.labels
+    }
+
+    fn update_proposal_labels(&mut self, proposal_id: ProposalId, new_labels: HashSet<String>) -> Promise {
+        let proposal: Proposal = self
+            .proposals
+            .get(proposal_id.into())
+            .unwrap_or_else(|| panic!("Proposal id {} not found", proposal_id))
+            .into();
+
+        self.edit_proposal_internal(proposal_id, proposal.snapshot.body, new_labels)
+    }
+
+    fn get_linked_proposals_in_rfp(&self, rfp_id: RFPId) -> HashSet<ProposalId> {
+        let rfp: RFP = self.rfps.get(rfp_id.into()).unwrap().into();
+        rfp.snapshot.linked_proposals
+    }
+
+    fn change_linked_proposal_in_rfp(&mut self, rfp_id: RFPId, proposal_id: ProposalId, operation: bool) {
```

Comment:
`operation: bool` is not intuitive. If it is necessary, I would use an enum of operation variants.